pylon-core/src/main/kotlin/io/github/pylonmc/pylon/core/item/builder/ItemStackBuilder.kt

```
@@ -197,38 +363,482 @@ open class ItemStackBuilder private constructor(val stack: ItemStack) : ItemProv
+        fun pylonTool(material: Material, key: NamespacedKey, toolType: ToolType, durability: Boolean)
+                = pylonTool(ItemStack(material), key, toolType, durability)
+
+        /**
+         * Creates a new [ItemStack] for a [PylonItem] that can be used as a weapon.
+         * You must provide a value for `attack-damage`, `attack-speed` and `attack-durability-damage` in the [Settings][Settings.get] for the item.
+         *
+         * If `durability` is true, gives the item a max durability defined by `durability` in the [Settings][Settings.get].
+         * Otherwise, removes any existing durability.
+         *
+         * If `knockback` is true, gives the item knockback defined by `attack-knockback` in the [Settings][Settings.get].
+         * Otherwise, removes any existing attack knockback.
+         *
+         * If `disablesShield` is true, gives the item a shield disable time defined by `disable-shield-seconds` in the [Settings][Settings.get].
+         * If false, the item will not disable shields when used in attacking.
+         *
+         * `settings/example-weapon.yml`
+         * ```yml
+         * attack-damage: 6.0 # double
+         * attack-speed: 1.6 # double
+         * attack-durability-damage: 2 # integer
+         * attack-knockback: 0.4 # double
+         * disable-shield-seconds: 3.0 # float
+         * durability: 500 # integer
+         * ```
+         */
+        @JvmStatic
+        fun pylonWeapon(stack: ItemStack, key: NamespacedKey, durability: Boolean, knockback: Boolean, disablesShield: Boolean) = create(stack, key) { builder, settings ->
+            builder.weapon(
+                settings.getOrThrow("attack-damage", ConfigAdapter.DOUBLE),
+                settings.getOrThrow("attack-speed", ConfigAdapter.DOUBLE),
+                settings.getOrThrow("attack-durability-damage", ConfigAdapter.INT),
+                if (disablesShield) settings.getOrThrow("disable-shield-seconds", ConfigAdapter.FLOAT) else 0f
```

if they specify that it should disable shields but don't give a time, I think that should be a throw not a silent fail